### PR TITLE
updating for node 4 and streams3 changes

### PR DESCRIPTION
**Pull request by dustinsmith1024**
~~@doug-martin @aheuermann don't merge yet, but this fixes the tests. Moving on to testing with C2FO and will report back.~~

Things are working. 

- [x] @doug-martin 
- [ ] @aheuermann

### CODE REVIEW

**Comment by dustinsmith1024** [comment]
So I think this is working well. I do think we should officially deprecate the 'record' event though.

It caused me some problems in node 5. If you don't have a 'data' handler then you are forced to call `resume` or `pipe`.  So all our streams that had 'record' were not working correctly.

They have a good explanation out here: https://nodejs.org/api/stream.html#stream_compatibility_with_older_node_js_versions

**Comment by doug-martin** [comment]
I agree and have wanted to deprecate it. This is a good opportunity to do it.

**Comment by dustinsmith1024** [comment]
@doug-martin just delete this line https://github.com/C2FO/fast-csv/blob/98951c16330c8dc95893c0e66a57b8163f77dfc9/lib/parser/parser_stream.js#L187?

**Comment by doug-martin** [comment]
Yep

**Comment by dustinsmith1024** [comment]
@doug-martin @aheuermann I think this one is good to go.  Any issues with me merging? I ran `grunt` to rebuild any docs. Anything else to do?

File: test/formatting.test.js [hunk 1]
```diff
@@ -460,57 +460,6 @@ it.describe("fast-csv formatting", function (it) {
                 });
             });
 
-            it.describe("if quoteColumns object and quoteHeaders is false it should only quote the specified column and not the header", function (it) {
```
Comment (doug-martin) [inline test/formatting.test.js]: Why did you remove these tests?

**Comment by doug-martin** [comment]
Can you update the history.md with a description of your changes also?

File: test/formatting.test.js (hunk 1 above)
Comment (dustinsmith1024) [inline test/formatting.test.js]: As far as I could tell its a duplicate of a test above it.  There was only 1 wording difference on the `work with multidimensional` part. 

https://github.com/C2FO/fast-csv/blob/node-5/test/formatting.test.js#L361

Comment (doug-martin) [inline test/formatting.test.js]: One has the `quoteColumns` option the other has `quoteHeaders`

Comment (dustinsmith1024) [inline test/formatting.test.js]: I think they are the same test. I am not seeing a difference. If I look at master and copy paste search they are duplicates.

**Comment by dustinsmith1024** [comment]
`Can you update the history.md with a description of your changes also?` - Done
